From e2e2f964b34b107c1d3be94b193a389aca1ba80c Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Tue, 19 Sep 2023 11:48:48 +0200
Subject: [PATCH] Don't populate list of keyboard layouts in the wayland
 greeter

Layout switching is not implemented. We should keep the list of
layouts empty so that themes can know to not show the option to
change layout
---
 src/greeter/waylandkeyboardbackend.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/greeter/waylandkeyboardbackend.cpp b/src/greeter/waylandkeyboardbackend.cpp
index c3cf460fc..fcea53614 100644
--- a/src/greeter/waylandkeyboardbackend.cpp
+++ b/src/greeter/waylandkeyboardbackend.cpp
@@ -92,7 +92,9 @@ QList<QObject *> parseRules(const QString &filename, int &current)
 
 void WaylandKeyboardBackend::init()
 {
-    d->layouts = parseRules(QStringLiteral("/usr/share/X11/xkb/rules/evdev.xml"), d->layout_id);
+    // TODO: We can't actually switch keyboard layout yet, so don't populate a list of layouts
+    // so that themes can know to not show the option to change layout
+    // d->layouts = parseRules(QStringLiteral("/usr/share/X11/xkb/rules/evdev.xml"), d->layout_id);
 }
 
 void WaylandKeyboardBackend::disconnect()
